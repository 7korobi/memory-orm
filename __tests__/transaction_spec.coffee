require "./models/index"
Mem = require "../src/index"

test_is = (target, list, key, src)->
  test "#{target} #{list} #{key}", ->
    base = Mem.State.base { list }
    tgt = JSON.parse JSON.stringify base[target][key]
    expect( tgt ).toEqual src

do_test = (list_id,{ $sort, $memory, $format })->
  test "$sort #{list_id}", ->
    expect( $sort ).toMatchSnapshot()
  test "$format #{list_id}", ->
    expect( $format ).toMatchSnapshot()
  test "$memory #{list_id}", ->
    expect( $memory ).toMatchSnapshot()

  for key, o of $sort
    test_is "$sort", list_id, key, o
  for key, o of $format
    test_is "$format", list_id, key, o
  for key, o of $memory
    test_is "$memory", list_id, key, o


describe "transaction", ->
  test "_step", ->
    expect( Mem.State.step ).toMatchSnapshot()
  undefined

describe "stringify", ->
  json = JSON.parse JSON.stringify Mem.Query.static.meta
  do_test "randoms", json.pack.randoms
  do_test "faces",   json.pack.faces
  do_test "tags",    json.pack.tags
  undefined

describe "json", ->
  json = JSON.parse Mem.Query.static.meta.json()
  do_test "randoms", json.pack.randoms
  do_test "faces",   json.pack.faces
  do_test "tags",    json.pack.tags
  undefined
