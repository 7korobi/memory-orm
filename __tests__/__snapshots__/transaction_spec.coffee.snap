// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transaction $format chr_jobs 1`] = `
Object {
  "_reduce": Object {
    "hash": Object {},
    "list": Array [],
  },
}
`;

exports[`transaction $format chr_npcs 1`] = `
Object {
  "_reduce": Object {
    "hash": Object {},
    "list": Array [],
  },
}
`;

exports[`transaction $format chr_sets 1`] = `
Object {
  "_reduce": Object {
    "hash": Object {},
    "list": Array [],
  },
}
`;

exports[`transaction $format faces 1`] = `
Object {
  "_reduce": Object {
    "count": 0,
    "hash": Object {},
    "list": Array [],
  },
  "_reduce.name_head.ア": Object {
    "hash": Object {},
  },
  "_reduce.name_head.イ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ウ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.エ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.オ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.カ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ガ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.キ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ギ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ク": Object {
    "hash": Object {},
  },
  "_reduce.name_head.グ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ケ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ゲ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.コ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ゴ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.サ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ザ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.シ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ジ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ス": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ズ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.セ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ゼ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ソ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ゾ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.タ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ダ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.チ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ツ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.テ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.デ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ト": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ド": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ナ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ニ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ヌ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ネ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ノ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ハ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.バ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.パ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ヒ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ビ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ピ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.フ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ブ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.プ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ヘ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ベ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ペ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ホ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ボ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ポ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.マ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ミ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ム": Object {
    "hash": Object {},
  },
  "_reduce.name_head.メ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.モ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ヤ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ユ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ヨ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ラ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.リ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ル": Object {
    "hash": Object {},
  },
  "_reduce.name_head.レ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ロ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ワ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ン": Object {
    "hash": Object {},
  },
  "_reduce.name_head.ヴ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.一": Object {
    "hash": Object {},
  },
  "_reduce.name_head.丁": Object {
    "hash": Object {},
  },
  "_reduce.name_head.亀": Object {
    "hash": Object {},
  },
  "_reduce.name_head.仁": Object {
    "hash": Object {},
  },
  "_reduce.name_head.余": Object {
    "hash": Object {},
  },
  "_reduce.name_head.八": Object {
    "hash": Object {},
  },
  "_reduce.name_head.勝": Object {
    "hash": Object {},
  },
  "_reduce.name_head.勢": Object {
    "hash": Object {},
  },
  "_reduce.name_head.博": Object {
    "hash": Object {},
  },
  "_reduce.name_head.参": Object {
    "hash": Object {},
  },
  "_reduce.name_head.団": Object {
    "hash": Object {},
  },
  "_reduce.name_head.夕": Object {
    "hash": Object {},
  },
  "_reduce.name_head.大": Object {
    "hash": Object {},
  },
  "_reduce.name_head.奈": Object {
    "hash": Object {},
  },
  "_reduce.name_head.定": Object {
    "hash": Object {},
  },
  "_reduce.name_head.小": Object {
    "hash": Object {},
  },
  "_reduce.name_head.志": Object {
    "hash": Object {},
  },
  "_reduce.name_head.慶": Object {
    "hash": Object {},
  },
  "_reduce.name_head.攻": Object {
    "hash": Object {},
  },
  "_reduce.name_head.日": Object {
    "hash": Object {},
  },
  "_reduce.name_head.明": Object {
    "hash": Object {},
  },
  "_reduce.name_head.春": Object {
    "hash": Object {},
  },
  "_reduce.name_head.朔": Object {
    "hash": Object {},
  },
  "_reduce.name_head.朝": Object {
    "hash": Object {},
  },
  "_reduce.name_head.朧": Object {
    "hash": Object {},
  },
  "_reduce.name_head.梅": Object {
    "hash": Object {},
  },
  "_reduce.name_head.櫻": Object {
    "hash": Object {},
  },
  "_reduce.name_head.沙": Object {
    "hash": Object {},
  },
  "_reduce.name_head.沼": Object {
    "hash": Object {},
  },
  "_reduce.name_head.法": Object {
    "hash": Object {},
  },
  "_reduce.name_head.源": Object {
    "hash": Object {},
  },
  "_reduce.name_head.玖": Object {
    "hash": Object {},
  },
  "_reduce.name_head.甚": Object {
    "hash": Object {},
  },
  "_reduce.name_head.直": Object {
    "hash": Object {},
  },
  "_reduce.name_head.稲": Object {
    "hash": Object {},
  },
  "_reduce.name_head.竹": Object {
    "hash": Object {},
  },
  "_reduce.name_head.網": Object {
    "hash": Object {},
  },
  "_reduce.name_head.置": Object {
    "hash": Object {},
  },
  "_reduce.name_head.舷": Object {
    "hash": Object {},
  },
  "_reduce.name_head.芙": Object {
    "hash": Object {},
  },
  "_reduce.name_head.菊": Object {
    "hash": Object {},
  },
  "_reduce.name_head.華": Object {
    "hash": Object {},
  },
  "_reduce.name_head.藤": Object {
    "hash": Object {},
  },
  "_reduce.name_head.辰": Object {
    "hash": Object {},
  },
  "_reduce.name_head.鉄": Object {
    "hash": Object {},
  },
  "_reduce.name_head.錠": Object {
    "hash": Object {},
  },
  "_reduce.name_head.鏡": Object {
    "hash": Object {},
  },
  "_reduce.name_head.雪": Object {
    "hash": Object {},
  },
  "_reduce.name_head.雷": Object {
    "hash": Object {},
  },
  "_reduce.name_head.露": Object {
    "hash": Object {},
  },
  "_reduce.name_head.鬼": Object {
    "hash": Object {},
  },
  "_reduce.name_head.麻": Object {
    "hash": Object {},
  },
  "_reduce.name_head.黍": Object {
    "hash": Object {},
  },
  "_reduce.name_head.Ａ": Object {
    "hash": Object {},
  },
  "_reduce.name_head.Ｐ": Object {
    "hash": Object {},
  },
  "_reduce.tag.all": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.animal": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.apartment": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.asia": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.ecclesia": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.elder": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.elegant": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.fable": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.farm": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.giji": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.god": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.government": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.guild": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.immoral": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.kid": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.marchen": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.market": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.medical": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.middle": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.myth": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.office": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.river": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.road": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.school": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.servant": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.shoji": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.stratos": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.student": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.travel": Object {
    "count": 0,
    "hash": Object {},
  },
  "_reduce.tag.young": Object {
    "count": 0,
    "hash": Object {},
  },
}
`;

exports[`transaction $format games 1`] = `
Object {
  "_reduce": Object {
    "hash": Object {},
    "list": Array [],
  },
}
`;

exports[`transaction $format locales 1`] = `
Object {
  "_reduce": Object {
    "hash": Object {},
    "list": Array [],
  },
}
`;

exports[`transaction $format marks 1`] = `
Object {
  "_reduce": Object {
    "hash": Object {},
    "list": Array [],
  },
}
`;

exports[`transaction $format options 1`] = `
Object {
  "_reduce": Object {
    "hash": Object {},
    "list": Array [],
  },
}
`;

exports[`transaction $format randoms 1`] = `
Object {
  "_reduce": Object {
    "hash": Object {},
    "list": Array [],
  },
  "_reduce.ratio": Object {
    "all": 0,
    "count": 0,
  },
  "_reduce.type.chess": Object {
    "count": 0,
  },
  "_reduce.type.coin": Object {
    "count": 0,
  },
  "_reduce.type.planet": Object {
    "count": 0,
  },
  "_reduce.type.tarot": Object {
    "count": 0,
  },
  "_reduce.type.trump": Object {
    "count": 0,
  },
  "_reduce.type.weather": Object {
    "count": 0,
  },
  "_reduce.type.zodiac": Object {
    "count": 0,
  },
}
`;

exports[`transaction $format says 1`] = `
Object {
  "_reduce": Object {
    "hash": Object {},
    "list": Array [],
  },
}
`;

exports[`transaction $format tags 1`] = `
Object {
  "_reduce": Object {
    "hash": Object {},
    "list": Array [],
  },
  "_reduce.group.1": Object {
    "hash": Object {},
    "list": Array [],
  },
  "_reduce.group.10": Object {
    "hash": Object {},
    "list": Array [],
  },
  "_reduce.group.2": Object {
    "hash": Object {},
    "list": Array [],
  },
  "_reduce.group.4": Object {
    "hash": Object {},
    "list": Array [],
  },
  "_reduce.group.5": Object {
    "hash": Object {},
    "list": Array [],
  },
  "_reduce.group.6": Object {
    "hash": Object {},
    "list": Array [],
  },
  "_reduce.group.7": Object {
    "hash": Object {},
    "list": Array [],
  },
  "_reduce.group.8": Object {
    "hash": Object {},
    "list": Array [],
  },
  "_reduce.group.9": Object {
    "hash": Object {},
    "list": Array [],
  },
}
`;

exports[`transaction $format winners 1`] = `
Object {
  "_reduce": Object {
    "hash": Object {},
    "list": Array [],
  },
}
`;

exports[`transaction $sort chr_jobs 1`] = `
Object {
  "_reduce.list": Object {
    "sort": Array [
      "face.order",
    ],
  },
}
`;

exports[`transaction $sort chr_npcs 1`] = `
Object {
  "_reduce.list": Object {},
}
`;

exports[`transaction $sort chr_sets 1`] = `
Object {
  "_reduce.list": Object {},
}
`;

exports[`transaction $sort faces 1`] = `
Object {
  "_reduce.list": Object {
    "sort": Array [
      "order",
    ],
  },
  "_reduce.name_head": Object {
    "cover": Array [
      "ア",
      "ィ",
      "イ",
      "ゥ",
      "ウ",
      "ェ",
      "エ",
      "ォ",
      "オ",
      "カ",
      "ガ",
      "キ",
      "ギ",
      "ク",
      "グ",
      "ケ",
      "ゲ",
      "コ",
      "ゴ",
      "サ",
      "ザ",
      "シ",
      "ジ",
      "ス",
      "ズ",
      "セ",
      "ゼ",
      "ソ",
      "ゾ",
      "タ",
      "ダ",
      "チ",
      "ヂ",
      "ッ",
      "ツ",
      "ヅ",
      "テ",
      "デ",
      "ト",
      "ド",
      "ナ",
      "ニ",
      "ヌ",
      "ネ",
      "ノ",
      "ハ",
      "バ",
      "パ",
      "ヒ",
      "ビ",
      "ピ",
      "フ",
      "ブ",
      "プ",
      "ヘ",
      "ベ",
      "ペ",
      "ホ",
      "ボ",
      "ポ",
      "マ",
      "ミ",
      "ム",
      "メ",
      "モ",
      "ャ",
      "ヤ",
      "ュ",
      "ユ",
      "ョ",
      "ヨ",
      "ラ",
      "リ",
      "ル",
      "レ",
      "ロ",
      "ヮ",
      "ワ",
      "ヰ",
      "ヱ",
      "ヲ",
      "ン",
    ],
    "index": "set.length",
    "sort": Array [
      "id",
    ],
  },
}
`;

exports[`transaction $sort games 1`] = `
Object {
  "_reduce.list": Object {},
}
`;

exports[`transaction $sort locales 1`] = `
Object {
  "_reduce.list": Object {},
}
`;

exports[`transaction $sort marks 1`] = `
Object {
  "_reduce.list": Object {},
}
`;

exports[`transaction $sort options 1`] = `
Object {
  "_reduce.list": Object {},
}
`;

exports[`transaction $sort randoms 1`] = `
Object {
  "_reduce.list": Object {
    "sort": Array [
      "ratio",
      "desc",
    ],
  },
  "_reduce.type": Object {
    "sort": Array [
      "count",
      "asc",
    ],
  },
}
`;

exports[`transaction $sort says 1`] = `
Object {
  "_reduce.list": Object {},
}
`;

exports[`transaction $sort tags 1`] = `
Object {
  "_reduce.group.1.list": Object {
    "sort": Array [
      "order",
    ],
  },
  "_reduce.group.10.list": Object {
    "sort": Array [
      "order",
    ],
  },
  "_reduce.group.2.list": Object {
    "sort": Array [
      "order",
    ],
  },
  "_reduce.group.4.list": Object {
    "sort": Array [
      "order",
    ],
  },
  "_reduce.group.5.list": Object {
    "sort": Array [
      "order",
    ],
  },
  "_reduce.group.6.list": Object {
    "sort": Array [
      "order",
    ],
  },
  "_reduce.group.7.list": Object {
    "sort": Array [
      "order",
    ],
  },
  "_reduce.group.8.list": Object {
    "sort": Array [
      "order",
    ],
  },
  "_reduce.group.9.list": Object {
    "sort": Array [
      "order",
    ],
  },
  "_reduce.list": Object {
    "sort": Array [
      "order",
    ],
  },
}
`;

exports[`transaction $sort winners 1`] = `
Object {
  "_reduce.list": Object {},
}
`;

exports[`transaction _step 1`] = `
Object {
  "ables": 761,
  "books": 762,
  "cards": 754,
  "chats": 809,
  "chr_jobs": 819,
  "chr_npcs": 818,
  "chr_sets": 817,
  "faces": 816,
  "folders": 791,
  "games": 825,
  "icons": 811,
  "locales": 820,
  "markers": 810,
  "marks": 824,
  "options": 821,
  "parts": 781,
  "phases": 795,
  "potofs": 804,
  "randoms": 814,
  "roles": 759,
  "says": 823,
  "sections": 783,
  "sow_roletables": 793,
  "sow_turns": 787,
  "sow_village_plans": 786,
  "sow_villages": 788,
  "stats": 755,
  "tags": 815,
  "winners": 822,
  "work_countrys": 812,
  "work_names": 813,
}
`;
